python/tank/pipelineconfig.py

```diff
@@ -83,59 +81,118 @@ def __init__(self, pipeline_configuration_path):
     def __repr__(self):
         return "<Sgtk Configuration %s>" % self._pc_root
 
-    def _clear_cached_settings(self):
+    ########################################################################################
+    # handling pc metadata
+    
+    def _get_metadata(self):
         """
-        Force the pc object to reread its settings from disk.
-        Call this if you have made changes to config files and 
-        want these to be picked up. The next time settings are needed,
-        these will be automatically re-read from disk.
+        Loads the pipeline config metadata (the pipeline_configuration.yml) file from disk.
+        
+        :param pipeline_config_path: path to a pipeline configuration root folder
+        :returns: deserialized content of the file in the form of a dict.
         """
-        self._project_id = self._UNDEFINED_PROJECT_ID
-        self._pc_id = None
-        self._pc_name = None
-        self._published_file_entity_type = None
-        self._cache_folder = None
-        self._path_cache_path = None
-        self._use_shotgun_path_cache = None
-
-    def _load_metadata_from_sg(self):
+    
+        # now read in the pipeline_configuration.yml file
+        cfg_yml = os.path.join(self.get_config_location(), 
+                               "core", 
+                               "pipeline_configuration.yml")
+    
+        if not os.path.exists(cfg_yml):
+            raise TankError("Configuration metadata file '%s' missing! "
+                            "Please contact support." % cfg_yml)
+    
+        fh = open(cfg_yml, "rt")
+        try:
+            data = yaml.load(fh)
+            if data is None:
+                raise Exception("File contains no data!")
+        except Exception, e:
+            raise TankError("Looks like a config file is corrupt. Please contact "
+                            "support! File: '%s' Error: %s" % (cfg_yml, e))
+        finally:
+            fh.close()
+    
+        return data
+    
+    def _update_pipeline_config(self, updates):
         """
-        Caches PC metadata from shotgun.
+        Updates the pipeline configuration on disk with the passed in values.
+
+        :param updates: Dictionary of values to update in the pipeline configuration
         """
-        sg = shotgun.get_sg_connection()
-        platform_lookup = {"linux2": "linux_path", "win32": "windows_path", "darwin": "mac_path" }
-        sg_path_field = platform_lookup[sys.platform]
-        data = sg.find_one(constants.PIPELINE_CONFIGURATION_ENTITY,
-                           [[sg_path_field, "is", self._pc_root]],
-                           ["id", "project", "code"])
-        if data is None:
-            raise TankError("Cannot find a Pipeline configuration in Shotgun that has its %s "
-                            "set to '%s'!" % (sg_path_field, self._pc_root))
+        # get current settings
+        curr_settings = self._get_metadata()
+        
+        # add path cache setting
```

Comment:
This comment seems to know in advance what the specific updates are? Is that right?